test/nlp_tests.jl

```diff
@@ -0,0 +1,23 @@
+@testset "JuMP NLP models" begin
+    @test MOI.supports(POI.ParametricOptimizer(Ipopt.Optimizer()), MOI.NLPBlock())
+
+    model = Model(() -> POI.ParametricOptimizer(Ipopt.Optimizer()))
+
+    @variable(model, x[i=1:2] >= 0)
+    @variable(model, y in POI.Parameter(10))
+    @constraint(model, sum(x) + y >= 0)
+    @NLobjective(model, Min, x[1]^2 + x[2]^2)
+    optimize!(model)
+end
+
+@testset "JuMP direct mode NLP models" begin
+    @test MOI.supports(POI.ParametricOptimizer(Ipopt.Optimizer()), MOI.NLPBlock())
+
+    optimizer = POI.ParametricOptimizer(Ipopt.Optimizer())
+    model = direct_model(optimizer)
+    @variable(model, x[i=1:2] >= 0)
+    @variable(model, y in POI.Parameter(10))
+    @constraint(model, sum(x) + y >= 0)
+    @NLobjective(model, Min, x[1]^2 + x[2]^2)
```

Comment:
what happens if you have a parameter on the NLPBlock?

Reply:
This shouldn't be allowed. There is already NPparameter

Reply:
ParameterJuMP allows the parameters as long as they are in the linear constraints